### PR TITLE
WiP: new feature: Add method to get all expected binary files of a compilation unit

### PR DESCRIPTION
**Pull request by msteinbeck**
This PR adds a new method to get all expected binary files of a compilation unit and fixes the computation of deeply nested types.

### CODE REVIEW

File: src/main/java/spoon/support/reflect/cu/CompilationUnitImpl.java [hunk 1]
```diff
@@ -162,51 +163,75 @@ public void setFile(File file) {
...
+		final String name = nameOfParent == null || nameOfParent.isEmpty()
+				? type.getSimpleName()
+				: nameOfParent + "$" + type.getSimpleName();
+		binaries.add(new File(baseDir, name + ".class"));
+		// Use 'getElements()' rather than 'getNestedTypes()' to also fetch
+		// anonymous types.
+		type.getElements(new TypeFilter<>(CtType.class)).stream()
+				// Exclude 'type' itself.
+				.filter(inner -> !inner.equals(type))
+				// Include only direct inner types.
```
[the remark this answers is not in the document]
Comment (pvojtechovsky) [reply src/main/java/spoon/support/reflect/cu/CompilationUnitImpl.java]: ok, I see. You call it recursively.

File: src/main/java/spoon/reflect/cu/CompilationUnit.java [hunk 2]
```diff
@@ -67,6 +67,13 @@
 	 */
 	void setLineSeparatorPositions(int[] lineSeparatorPositions);
 
+	/**
+	 * Gets all binary (.class) files that corresponds to this compilation unit
+	 * and should be available after calling
+	 * {@link spoon.SpoonModelBuilder#compile(spoon.SpoonModelBuilder.InputType...)}.
+	 */
+	List<File> getExpectedBinaryFiles();
```
Comment (monperrus) [inline src/main/java/spoon/reflect/cu/CompilationUnit.java]: I would propose to put this public method in `SpoonModelBuilder` instead.

Comment (msteinbeck) [reply src/main/java/spoon/reflect/cu/CompilationUnit.java]: This method is specific to a CompilationUnit. The returned files correspond to the CompilationUnit the method was called on.

File: src/main/java/spoon/support/reflect/cu/CompilationUnitImpl.java [hunk 3]
```diff
@@ -162,51 +163,75 @@ public void setFile(File file) {
...
-								binaries.add(fileOfInnerType);
-							}
-						}
-					}
-				}
-			}
+			getDeclaredTypes().forEach(type ->
+					getExpectedBinaryFiles(base, null, type)
+							.stream()
+							.filter(File::isFile)
```
[the remark this answers is not in the document]
Comment (msteinbeck) [reply src/main/java/spoon/support/reflect/cu/CompilationUnitImpl.java]: Apparently, this is a copy paste error. Fixed it in latest commit.

**Comment by monperrus** [comment]
conflicts here, closing this stalled PR?

**Comment by monperrus** [comment]
ok, thanks!

**Pull request closed without merging**